pkg/segment/writer/startree_test.go

```diff
@@ -769,6 +770,61 @@ func getTotalLevels(node *Node) int {
 	return 1 + max
 }
 
+func (builder *StarTreeBuilder) DeriveEnc(wip WipBlock, recGroupByValues [][]string) (map[string]map[string]uint32, error) {
+	numRecs := wip.blockSummary.RecCount
+	mp := make(map[string]map[string]uint32)
+	for colNum := range builder.groupByKeys {
+		mp[builder.groupByKeys[colNum]] = make(map[string]uint32)
+	}
+
+	for recNum := uint16(0); recNum < numRecs; recNum += 1 {
+		for i, grpValues := range recGroupByValues[recNum] {
+			mp[builder.groupByKeys[i]][grpValues] = builder.wipRecNumToColEnc[i][recNum]
+		}
+	}
+
+	return mp, nil
+}
+
+func getTree(t *testing.T, root *Node, ans map[string][]utils.CValueEnclosure, key string) {
+	var newKey string
+	if key == "" {
+		newKey = fmt.Sprintf("%v", root.myKey)
+	} else {
+		newKey = strings.Join([]string{key, fmt.Sprintf("%v", root.myKey)}, "_")
+	}
+	if root.children == nil || len(root.children) == 0 {
+		ans[newKey] = root.aggValues
+		return
+	}
+
+	for _, child := range root.children {
+		getTree(t, child, ans, newKey)
+	}
+}
+
+func createKey(root *Node, encMap map[string]map[string]uint32, grpCols []string, grpVals []string) string {
+	keys := []string{fmt.Sprintf("%v", root.myKey)}
+	for i, col := range grpCols {
+		keys = append(keys, fmt.Sprintf("%v", encMap[col][grpVals[i]]))
+	}
+	return strings.Join(keys, "_")
+}
+
+func testAggs(t *testing.T, expected []int, aggVals []utils.CValueEnclosure) {
+	assert.Equal(t, len(expected), len(aggVals))
+
+	for i := 0; i < len(expected); i++ {
+		if i%TotalMeasFns == MeasFnCountIdx {
```

Comment:
nit: it looks like this function implicitly assumes `expected` (and `aggVals`) are for [these](https://github.com/siglens/siglens/blob/develop/pkg/segment/writer/agiletree.go#L42-L45) measure functions, in exactly that order. It might be good to make that assumption more explicit

Reply:
addressed in the latest commit.